src/category_theory/limits/shapes/binary_products.lean

```diff
@@ -255,12 +288,24 @@ abbreviation prod.map {W X Y Z : C} [has_limits_of_shape.{v} (discrete walking_p
   (f : W ⟶ Y) (g : X ⟶ Z) : W ⨯ X ⟶ Y ⨯ Z :=
 lim.map (map_pair f g)
 
+/-- If the products `W ⨯ X` and `Y ⨯ Z` exist, then every pair of isomorphisms `f : W ≅ Y` and
+    `g : X ≅ Z` induces a isomorphism `prod.map_iso f g : W ⨯ X ≅ Y ⨯ Z`. -/
+abbreviation prod.map_iso {W X Y Z : C} [has_limits_of_shape.{v} (discrete walking_pair) C]
+  (f : W ≅ Y) (g : X ≅ Z) : W ⨯ X ≅ Y ⨯ Z :=
+lim.map_iso (map_pair_iso f g)
+
```

Comment:
What's the advantage of making this an abbreviation rather than a def? In either case, it would be nice to have simp lemmas expressing `(prod.map_iso f g).hom = prod.map f.hom g.hom` etc to make it easier to work with this new definition in practice.

Reply:
First --- I'm very open to criticism of use of `abbreviation`! We don't do it much in mathlib, and I'm sure I'm often doing it wrong.

I think what you're asking for is precisely the advantage, however, of using `abbreviation` here. With `abbreviation` this works:
```
example {W X Y Z : C} [has_limits_of_shape.{v} (discrete walking_pair) C]
  (f : W ≅ Y) (g : X ≅ Z) : (prod.map_iso f g).hom = prod.map f.hom g.hom := by simp
```
while with `def` it doesn't (and we'd need to restate the `simp` lemma, proved by unfolding `prod.map_iso`).

Reply:
I can see it works in that example, but if I have `(prod.map_iso f g).hom` in the middle of a chain of compositions and I want to change it to `prod.map f.hom g.hom` (say, to use `prod.lift_map` or friends) then `simp` would take me too and convert it to `lim.map` instead, which would make everything harder. I haven't fully got a handle on what I think of abbreviation so I certainly won't complain about that; but I think explicitly stating this lemma that you've shown here would be helpful.

Reply:
Good point! Fixed.